src/engines/html5/html5.js

```diff
@@ -54,6 +54,12 @@ export default class Html5 extends FakeEventTarget implements IEngine {
    * @private
    */
   _canLoadMediaSourceAdapterPromise: Promise<*>;
+  /**
+   * Promise to indicate when a media source adapter can be loaded.
+   * @type {Promise<*>}
+   * @private
+   */
+  _registeredMetadataEvent: boolean;
```

Comment:
Do we really need this flag? can we rid of it in some way?

Reply:
Yes - since we want to make sure we try to find this track in every playback try

Reply:
So need to reset it on change media?